traits/ctraits.c

```diff
@@ -3505,12 +3515,13 @@ validate_trait_tuple_check(
                 }
                 else {
                     aitem = itrait->validate(itrait, obj, name, bitem);
-                }
-
-                if (aitem == NULL) {
-                    PyErr_Clear();
-                    Py_XDECREF(tuple);
-                    return NULL;
+                    if (aitem == NULL) {
```

Comment:
Note to reviewers: this exception-handling block was moved inside the `else`; this doesn't change behaviour (since there's no way that the `if` branch can result in `aitem` being `NULL`), but it improves the readability of the code a smidgen by bringing the exception handling closer to the `itrait->validate` call that the exception is being handled for.

Reply:
For the sake of clarity, I've reverted the refactor.